checks/webhook.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/checks/raw"
+	sce "github.com/ossf/scorecard/v4/errors"
+)
+
+const (
+	// CheckWebHooks is the registered name for WebHooks.
+	CheckWebHooks = "Webhooks"
+)
+
+//nolint:gochecknoinits
+func init() {
+	if err := registerCheck(CheckWebHooks, WebHooks, nil); err != nil {
+		// this should never happen
+		panic(err)
+	}
+}
+
+// WebHooks run Webhooks check.
+func WebHooks(c *checker.CheckRequest) checker.CheckResult {
+	rawData, err := raw.WebHook(c)
+	if err != nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, err.Error())
+		return checker.CreateRuntimeErrorResult(CheckWebHooks, e)
+	}
+
+	if len(rawData.Webhook) < 1 {
+		return checker.CreateMaxScoreResult(CheckWebHooks, "no webhooks defined")
+	}
+
+	hasNoSecretCount := 0
+	for _, hook := range rawData.Webhook {
+		if !hook.UsesAuthSecret {
+			c.Dlogger.Warn(&checker.LogMessage{
+				Path: fmt.Sprintf("https://%s/settings/hooks/%d", c.RepoClient.URI(), hook.ID),
+				Type: checker.FileTypeURL,
+				Text: "Webhook with no secret configured",
+			})
+			hasNoSecretCount++
+		}
+	}
+
+	generateWebhookTable(c.RepoClient.URI(), rawData.Webhook)
```

Comment:
you don't need this. Once the results are logged using `c.Dlogger.Warn` in the file `checks/evaluation/webhook.go`, the results will automatically be available in the scorecard reults: both the table, json, and sarif format.

Let me know if you have further questions.

Reply:
thanks!